File: onlinevote/src/main/java/com/example/vote/onlinevote/repository/UserRepository.java
package com.example.vote.onlinevote.repository;

import java.util.List;

import org.springframework.data.jpa.repository.JpaRepository;

import com.example.vote.onlinevote.model.User;

public interface UserRepository extends JpaRepository <User, Long>{

    List<User> findByRoleNot(String string);

}

File: Kotlin/testsecurity/src/main/kotlin/com/example/testsecurity/repository/UserRepository.kt
package com.example.testsecurity.repository

import com.example.testsecurity.model.User
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.Optional

@Repository
interface UserRepository : JpaRepository<User, Long> {
    
    // Find user by email (used for authentication)
    fun findByEmail(email: String): Optional<User>
    
    // Check if user exists by email
    fun existsByEmail(email: String): Boolean
}